app/src/main/java/io/homeassistant/companion/android/settings/language/LanguagesProvider.kt

```diff
@@ -0,0 +1,71 @@
+package io.homeassistant.companion.android.settings.language
+
+import android.content.Context
+import android.content.res.Resources
+import android.os.Build
+import android.os.LocaleList
+import android.util.DisplayMetrics
+import io.homeassistant.companion.android.BuildConfig
+import io.homeassistant.companion.android.R
+import java.util.Locale
+
+class LanguagesProvider {
+
+    fun getSupportedLanguages(context: Context): Map<String, String> {
+        val listAppLocales = mutableMapOf<String, String>()
+        val langManager = LanguagesManagerProvider().getManager(context)
+        val resources = context.resources
+
+        if (langManager.getAppVersion() != BuildConfig.VERSION_NAME || langManager.getLocales().isNullOrEmpty()) {
+            val listLocales = resources.assets.locales
+            val defString = getStringResource(context, "")
+            var supportedLocales = ""
+
+            listLocales.forEach {
```

Comment:
Would it make more sense to sort this by the display name? Not sure how it works in other languages but its odd to have it sorted by the language code to me.

Reply:
Probably you are right